apps/ewallet/lib/ewallet/validators/transaction_consumption_validator.ex

```diff
@@ -102,6 +97,28 @@ defmodule EWallet.TransactionConsumptionValidator do
     end
   end
 
+  defp validate_not_expired(consumption) do
+    case TransactionConsumption.expired?(consumption) do
+      false ->
+        {:ok, consumption}
+
+      true ->
+        Event.dispatch(:transaction_consumption_finalized, %{consumption: consumption})
```

Comment:
Why does it need to dispatch an event every time the validation failed?

Reply:
So in a case where `A` creates a transaction `request` with validation and `B` consumes it:
- `A` creates the request
- `B` consumes
- `A` receives a socket indicating that `B` wants to consume, `A` can then approve or reject this consumption
- `A` tries to approve it but it fails because the consumption is expired (This is when it comes in this function and try to validate his confirmation).
- `B` is notified about it that he doesn't need to wait for the consumption approval (maybe a `approval pending` screen was displayed?) because the consumption has expired.

Reply:
It was working like this until now, I just moved the logic in a function instead of being in the `with` clause because I needed to also check for `cancelled` in addition to `expired`.
So basically yes, it'll send an event every time `A` tries to approve the cancelled (or expired) request, but there is no reason for this. It's meant to be handled in the front end side that if `A` realized that the consumption is cancelled (or expired), the confirmation UI should be hidden.